src/app/proposals/hooks/useGetProposalsWithGraph.ts

```diff
@@ -1,24 +1,126 @@
 import { useMemo } from 'react'
 import { useBlockNumber } from 'wagmi'
-import { useQuery } from '@tanstack/react-query'
-import { getCachedProposals, ProposalGraphQLResponse } from '@/app/proposals/actions/proposalsAction'
+import { keepPreviousData, useQuery } from '@tanstack/react-query'
+import { useReadContracts } from 'wagmi'
 import { AVERAGE_BLOCKTIME } from '@/lib/constants'
 import Big from '@/lib/big'
-import { getProposalEventArguments, getProposalCategoryFromParsedData } from '@/app/proposals/shared/utils'
-import { Address, formatEther } from 'viem'
-import { ProposalState } from '@/shared/types'
+import { ProposalState, ProposalCategory } from '@/shared/types'
 import { Proposal } from '../shared/types'
+import { ProposalApiResponse } from '@/app/proposals/shared/types'
+import moment from 'moment'
+import { GovernorAbi } from '@/lib/abis/Governor'
+import { GOVERNOR_ADDRESS } from '@/lib/constants'
+import { formatEther } from 'viem'
+
+function proposalStateToRawState(proposalState: string): number {
+  const stateMap: Record<string, number> = {
+    Pending: ProposalState.Pending,
+    Active: ProposalState.Active,
+    Succeeded: ProposalState.Succeeded,
+    Defeated: ProposalState.Defeated,
+    Executed: ProposalState.Executed,
+    Canceled: ProposalState.Canceled,
+    Queued: ProposalState.Queued,
+    Expired: ProposalState.Expired,
+  }
+
+  return stateMap[proposalState] || ProposalState.Pending
+}
+
+async function fetchProposalsFromAPI(): Promise<ProposalApiResponse[]> {
+  console.log('fetching proposals from API')
+  const response = await fetch('/api/proposals/v1')
+  if (!response.ok) {
+    throw new Error(`Failed to fetch proposals: ${response.statusText}`)
+  }
+  return response.json()
+}
 
 export function useGetProposalsWithGraph() {
   const {
     data: proposalsData,
     isLoading: proposalDataIsLoading,
     error: proposalsDataError,
   } = useQuery({
-    queryFn: () => getCachedProposals(),
+    queryFn: fetchProposalsFromAPI,
     queryKey: ['proposals'],
     refetchInterval: AVERAGE_BLOCKTIME,
   })
+
+  const proposalsFromNode = useMemo(
+    () =>
+      proposalsData?.filter(
+        proposal => !proposal.votes || !proposal.proposalState || !proposal.quorumAtSnapshot,
+      ) || [],
+    [proposalsData],
+  )
+
+  const { data: proposalVotes } = useReadContracts({
```

Comment:
This is using 3 .map function and could be transformed into a single .reduce function.

Simply create a .reduce function, and return an object with:

votesContracts
quorumContracts
stateContracts

Then in each iteration, just create each contract payload, and push to each corresponding array.